lib/Sema/TypeChecker.h

```diff
@@ -756,7 +754,16 @@ void checkSwitchExhaustiveness(const SwitchStmt *stmt, const DeclContext *DC,
 /// \returns true if an error occurred, false otherwise.
 bool typeCheckCondition(Expr *&expr, DeclContext *dc);
 
-/// Type check the given 'if' or 'while' statement condition, which
+/// Type check the given 'if', 'while', or 'guard' statement condition.
+///
+/// \param stmt The conditional statement to type-check, which will be modified
+/// in place.
+///
+/// \returns true if an error occurred, false otherwise.
+bool typeCheckConditionForStatement(LabeledConditionalStmt *stmt,
```

Comment:
It looks like `typeCheckConditionForStatement` is now the only client of `typeCheckStmtCondition`. Should we merge it into the former?

Reply:
Yes. Let’s keep stripping TypeChecker.h please.

Reply:
Done!